frontend/src/app/shared/components/notifications/notifications.component.ts

```diff
@@ -1,15 +1,127 @@
-import { Component, Input } from '@angular/core';
+import { AfterViewInit, Component, ElementRef, HostListener, Input, OnInit, ViewChild } from '@angular/core';
 import { BaseComponent } from '@core/base/base.component';
+import { AuthService } from '@core/services/auth.service';
+import { EventService } from '@core/services/event.service';
+import { NotificationService } from '@core/services/notification.service';
+import { ToastrNotificationsService } from '@core/services/toastr-notifications.service';
+import { UserService } from '@core/services/user.service';
+import { FriendshipStatus } from '@shared/enums/friendship-status';
 import { TypeNotification } from '@shared/enums/type-notification';
+import { IUpdateFriendship } from '@shared/models/friendship/update-friendship';
 import { INotificationModel } from '@shared/models/notifications/notifications';
+import { IUser } from '@shared/models/user/user';
+import { takeUntil } from 'rxjs';
 
 @Component({
     selector: 'app-notifications',
     templateUrl: './notifications.component.html',
     styleUrls: ['./notifications.component.sass'],
 })
-export class NotificationsComponent extends BaseComponent {
-    @Input() notifications: INotificationModel[];
+export class NotificationsComponent extends BaseComponent implements OnInit, AfterViewInit {
+    constructor(
+        private notificationService: NotificationService,
+        private userService: UserService,
+        private authService: AuthService,
+        private toastrNotification: ToastrNotificationsService,
+        private eventService: EventService,
+    ) {
+        super();
+    }
+
+    private readonly TOP_OFFSET = 40;
+
+    private readonly RIGHT_OFFSET = 300;
+
+    private readonly HALF = 2;
+
+    private readonly NEGATIVE = -1;
+
+    @Input() public notifications: INotificationModel[];
+
+    @ViewChild('notificationsContainer') public notificationsContainer: ElementRef;
+
+    @HostListener('window:resize', ['$event'])
+    onResize() {
+        this.calculateRigthPosition();
+        this.calculateTopPosition();
+    }
+
+    public ngAfterViewInit() {
+        this.calculateTopPosition();
+        this.calculateRigthPosition();
+    }
+
+    public calculateTopPosition() {
+        const containerHeight = this.notificationsContainer.nativeElement.offsetHeight;
+        const topValue = containerHeight / this.HALF + this.TOP_OFFSET;
+
+        this.notificationsContainer.nativeElement.style.top = `${topValue}px`;
+    }
+
+    public calculateRigthPosition() {
+        const screenWidth = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
+        const rightValue = this.NEGATIVE * (screenWidth / this.HALF) + this.RIGHT_OFFSET;
+
+        this.notificationsContainer.nativeElement.style.right = `${rightValue}px`;
+    }
```

Comment:
I believe we don't need this all at all as this component should not be a modal window. this should be just a dropdown with a list of items

Reply:
For now, this component incorrectly shows notifications, when there are 4 or more of them. So we need to recalculate top margin to avoid this. 

## before 
![image](https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/assets/56970649/ff8149de-ae3e-447f-8478-8a547fd3fb89)

## after
![image](https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/assets/56970649/6174265f-9477-4c37-92c7-04f3442c7a92)

additionaly, calculateRigthPosition fix this:

## before 
![image](https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/assets/56970649/401c1348-b5fa-4acd-b25f-52eaa684a3d5)

## after (it always fixed under the bell on large screens)
![image](https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/assets/56970649/8d911be3-013e-49db-b350-be2344c32bd8)

Reply:
we have a card for redesigning this component. so we wont needed this in a future